editoast/src/views/v2/path/pathfinding.rs

```diff
@@ -186,27 +187,28 @@ async fn pathfinding_blocks_batch(
     Ok(pathfinding_results.into_iter().flatten().collect())
 }
 
-async fn build_pathfinding_request(
+fn build_pathfinding_request(
     pathfinding_input: &PathfindingInput,
-    conn: &mut DbConnection,
     infra: &Infra,
-) -> Result<std::result::Result<PathfindingRequest, PathfindingResult>> {
-    let path_items = &pathfinding_input.path_items;
+    path_item_cache: &PathItemCache,
+) -> std::result::Result<PathfindingRequest, PathfindingResult> {
+    let path_items: Vec<_> = pathfinding_input.path_items.iter().collect();
     if path_items.len() <= 1 {
-        return Ok(Err(PathfindingResult::NotEnoughPathItems));
+        return Err(PathfindingResult::NotEnoughPathItems);
     }
-    let track_offsets = match extract_location_from_path_items(conn, infra.id, path_items).await? {
-        Ok(track_offsets) => track_offsets,
-        Err(e) => return Ok(Err(e.into())),
-    };
+    let track_offsets = path_item_cache.extract_location_from_path_items(&path_items)?;
 
     // Check if tracks exist
-    if let Err(err) = check_tracks_from_path_items(conn, infra.id, &track_offsets).await? {
-        return Ok(Err(err));
+    for tracks in &track_offsets {
+        for track_offset in tracks {
+            if !path_item_cache.track_exists(&track_offset.track.0) {
+                return Err(PathfindingResult::NotFoundInTracks);
+            }
+        }
     }
```

Comment:
Something feels weird with that piece of code. You calculate `track_offsets` from `path_item_cache`. Then you iterate on all the track offsets to ask again to check if the track exists to `path_item_cache`. Shouldn't all of this be done directly in `extract_location_from_path_items`?

Reply:
I think it's linked to this comment: https://github.com/OpenRailAssociation/osrd/pull/8035#discussion_r1675588199

If we handle this in `extract_location_from_path_items`, we lose precision on the error type.